src/tools/miri/src/intrinsics/mod.rs

```diff
@@ -522,3 +573,80 @@ fn apply_random_float_error_to_imm<'tcx>(
 
     interp_ok(ImmTy::from_scalar_int(res, val.layout))
 }
+
+/// For the intrinsics:
+/// - sinf32, sinf64
+/// - cosf32, cosf64
+/// - expf32, expf64, exp2f32, exp2f64
+/// - logf32, logf64, log2f32, log2f64, log10f32, log10f64
+/// - powf32, powf64
+///
+/// Returns Some(`output`) if the `intrinsic` results in a defined fixed `output` specified in the C standard when given `args`
+/// as arguments, else None.
+fn fixed_float_value<S: Semantics>(
+    intrinsic_name: &str,
+    args: &[IeeeFloat<S>],
+) -> Option<IeeeFloat<S>> {
+    let one = IeeeFloat::<S>::one();
+    match (intrinsic_name, args) {
+        // sin(+- 0) = +- 0.
+        ("sinf32" | "sinf64", [input]) if input.is_zero() => Some(*input),
+
+        // cos(+- 0) = 1
+        ("cosf32" | "cosf64", [input]) if input.is_zero() => Some(one),
+
+        // e^0 = 1
+        ("expf32" | "expf64" | "exp2f32" | "exp2f64", [input]) if input.is_zero() => Some(one),
+
+        // log(1) = 0
+        #[rustfmt::skip]
+        ("logf32"
+        | "logf64"
+        | "log10f32"
+        | "log10f64"
+        | "log2f32"
+        | "log2f64", [input]) if *input == one => Some(IeeeFloat::<S>::ZERO),
+
+        // 1^y = 1 for any y, even a NaN.
+        ("powf32" | "powf64", [base, _]) if *base == one => Some(one),
+
+        // (-1)^(±INF) = 1
+        ("powf32" | "powf64", [base, exp]) if *base == -one && exp.is_infinite() => Some(one),
+
+        // x^(±0) = 1 for any x, even a NaN
+        ("powf32" | "powf64", [_, exp]) if exp.is_zero() => Some(one),
+
+        // C standard doesn't specify any fixed outputs for other combinations of `intrinsic_name` and `args`,
+        // or an invalid combination was given.
```

Comment:
Looking at C23, this does not seem true? There's like 20 cases just for `pow`.

However, we can skip all the cases that return infinity, since applying an error to infinity changes nothing. We could also skip all the cases that return 0, for the same reason (but currently we don't seem to do that consistently).